homeassistant/components/ambiclimate/config_flow.py

```diff
@@ -132,12 +132,12 @@ def _generate_oauth(self) -> ambiclimate.AmbiclimateOAuth:
             clientsession,
         )
 
-    def _cb_url(self):
+    def _cb_url(self) -> str:
         return f"{get_url(self.hass, prefer_external=True)}{AUTH_CALLBACK_PATH}"
 
-    async def _get_authorize_url(self):
+    async def _get_authorize_url(self) -> str:
         oauth = self._generate_oauth()
-        return oauth.get_authorize_url()
+        return oauth.get_authorize_url()  # type: ignore[no-any-return]
```

Comment:
```suggestion
        return str(oauth.get_authorize_url())
```

Reply:
Converting to `str` changes the code, and is not the same as typing.

Reply:
Yeah, you are right, then a cast should be better

Reply:
When assigned to a `str` types var, and returning that, mypy won't complain. I think in this case the type ignore is fine.